scipy/sparse/linalg/eigen/arpack/arpack.py

```diff
@@ -1231,11 +1231,30 @@ def eigs(A, k=6, M=None, sigma=None, which='LM', v0=None,
             import warnings
             warnings.warn('M does not have the same type precision as A. '
                           'This may adversely affect ARPACK convergence')
+
     n = A.shape[0]
 
-    if k <= 0 or k >= n:
-        raise ValueError("k=%d must be between 1 and ndim(A)-1=%d"
-                         % (k, n - 1))
+    if k <= 0:
+        raise ValueError("k=%d must be greater than 0." % (k, n - 1))
```

Comment:
You're giving too many values to the format string here. We aren't using `n - 1` anymore.